Source/WebCore/platform/graphics/cg/ImageBufferIOSurfaceBackend.cpp

```diff
@@ -239,6 +252,49 @@ void ImageBufferIOSurfaceBackend::ensureNativeImagesHaveCopiedBackingStore()
     flushContext();
 }
 
+void* ImageBufferIOSurfaceBackend::updateBitmapCacheForRead()
+{
+    if (!m_bitmapCacheLock.has_value()) {
+        flushContext();
+        m_bitmapCacheLock = IOSurface::Locker { *m_surface, IOSurface::Locker::AccessMode::ReadOnly };
+    }
+    return m_bitmapCacheLock->surfaceBaseAddress();
+}
+
+CGImageRef ImageBufferIOSurfaceBackend::updateBitmapCacheImage()
+{
+    if (m_bitmapCacheImage)
+        return m_bitmapCacheImage.get();
+    void* data = updateBitmapCacheForRead();
+    if (!data)
+        return nullptr;
+    auto configuration = m_surface->bitmapConfiguration();
+    auto size = m_surface->size();
+    // FIXME: we can cache also the context and just use set data operation, in later commits.
+    auto context = adoptCF(CGBitmapContextCreate(data, size.width(), size.height(), configuration.bitsPerComponent, m_surface->bytesPerRow(), m_surface->colorSpace().platformColorSpace(), configuration.bitmapInfo));
```

Comment:
> Did we measure how much speed progression we get from this change?
Yes, that's the idea of these changes.
So at the moment we are, for example:
- 160 for the incorrect rendering
- 130 for the current trunk, incorrect rendering fixed
- 170 when the fix sequence has landed

> Is there a memory regression because of this change?

None, that's what the CG does internally when the IOSurface-backed CGImage is drawn to bitmap canvas.